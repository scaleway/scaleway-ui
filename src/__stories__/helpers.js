export const showAlert = (text = 'fired!') => {
  // eslint-disable-next-line no-alert
  alert(text)
}

export const log = (text = 'fired!') => {
  // eslint-disable-next-line no-console
  console.log(text)
}
