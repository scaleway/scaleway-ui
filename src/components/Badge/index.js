import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import React from 'react'
import {
  white,
  success,
  beta,
  info,
  red,
  warning,
  gray700,
  gray350,
  bigBorderRadius,
  borderRadius,
} from '../../theming'
import { cx, sp } from '../../utils'
import { Box } from '../Box'

const variants = {
  success: p => css`
    background-color: ${success(p)};
  `,
  beta: p => css`
    background-color: ${beta(p)};
  `,
  warning: p => css`
    background-color: ${warning(p)};
  `,
  error: p => css`
    background-color: ${red(p)};
  `,
  info: p => css`
    background-color: ${info(p)};
  `,
  neutral: p => css`
    color: ${gray700(p)};
    background-color: ${gray350(p)};
  `,
}

const sizes = {
  rounded: p => {
    const height = sp(2)(p)
    const paddingVertical = sp(0.25)(p)
    const paddingHorizontal = sp(0.75)(p)
    return css`
      border-radius: ${bigBorderRadius(p)};
      font-size: 10px;
      height: ${height};
      padding: ${paddingVertical} ${paddingHorizontal};
      text-transform: uppercase;
    `
  },
  md: p => {
    const height = sp(4)(p)
    return css`
      font-size: 14px;
      line-height: ${height};
      height: ${height};
    `
  },
  sm: p => {
    const height = sp(3)(p)
    return css`
      font-size: 12px;
      line-height: ${height};
      height: ${height};
    `
  },
  xs: p => {
    const height = sp(2.25)(p)
    return css`
      border-radius: ${borderRadius(p)};
      font-size: 12px;
      line-height: ${height};
      height: ${height};
    `
  },
  xxs: p => {
    const height = sp(2)(p)
    const padding = sp(0.75)(p)
    return css`
      border-radius: ${borderRadius(p)};
      font-size: 10px;
      line-height: ${height};
      height: ${height};
      padding: 0 ${padding};
    `
  },
}

const style = p => css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  color: ${white(p)};
  white-space: nowrap;
  border-radius: ${sp(2)(p)};
  padding: 0 ${sp(2)(p)};
  width: fit-content;
`

const Badge = ({ variant = 'neutral', size = 'md', ...props }) => (
  <Box as="span" css={cx([style, variants[variant], sizes[size]])} {...props} />
)

Badge.propTypes = {
  variant: PropTypes.oneOf(Object.keys(variants)),
  size: PropTypes.oneOf(Object.keys(sizes)),
}

Badge.defaultProps = {
  variant: 'neutral',
  size: 'md',
}

export { Badge }
