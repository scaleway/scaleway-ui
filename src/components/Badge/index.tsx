import { Theme, useTheme } from '@emotion/react'
import styled from '@emotion/styled'
import { ReactNode, useMemo } from 'react'
import { Color, SENTIMENTS_WITHOUT_NEUTRAL } from '../../theme'
import capitalize from '../../utils/capitalize'
import Icon, { IconName } from '../Icon'
import Typography from '../Typography'

const StyledTypography = styled(Typography)`
  text-transform: uppercase;
  font-size: ${({ fontSize }) => fontSize}px;
`

// TODO: remove when typography has been created
const TEXT_SIZES = {
  large: 14,
  medium: 12,
  small: 10,
}

export const SIZES = {
  large: 32,
  medium: 24,
  small: 16,
}

export const PROMINENCES = {
  default: 'default',
  strong: 'strong',
}

/**
 * Generate all styles available for badge based on prominence and variants
 * @param prominence
 * @param theme
 */
const generateStyles = ({
  prominence,
  theme,
}: {
  prominence: string
  theme: Theme
}): Record<string, string> => {
  const definedProminence =
    prominence === PROMINENCES.strong ? capitalize(PROMINENCES.strong) : ''

  const text = `text${definedProminence}`
  const background = `background${definedProminence}`

  return {
    ...SENTIMENTS_WITHOUT_NEUTRAL.reduce(
      (reducer, sentiment) => ({
        ...reducer,
        [sentiment]: `
      color: ${
        theme.colors[sentiment][text as keyof typeof theme.colors.primary]
      };
      background: ${
        theme.colors[sentiment][background as keyof typeof theme.colors.primary]
      }
    `,
      }),
      {},
    ),
    disabled: `
      color: ${theme.colors.neutral.textWeak};
      background: ${theme.colors.neutral.backgroundStrong}
    `,
    neutral: `
      color: ${
        prominence === PROMINENCES.strong
          ? theme.colors.neutral[text as keyof typeof theme.colors.neutral]
          : theme.colors.neutral.textWeak
      };
      background: ${
        theme.colors.neutral[background as keyof typeof theme.colors.neutral]
      }
    `,
  }
}

const StyledBox = styled('div', {
  shouldForwardProp: prop => !['variant', 'size'].includes(prop.toString()),
})<{ size: number; variant: string }>`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: ${({ theme }) => theme.space['2']};
  padding: 0
    ${({ theme, size }) =>
      size === SIZES.small ? theme.space['1'] : theme.space['2']};
  gap: ${({ theme, size }) =>
    size === SIZES.small ? theme.space['0.5'] : theme.space['1']};
  width: fit-content;
  height: ${({ size }) => size}px;
  ${({ variant }) => variant}
`

type BadgeProps = {
  variant?: Color
  size?: keyof typeof SIZES
  prominence?: keyof typeof PROMINENCES
  /**
   * Defines icon to display on left side of badge. **Only available on medium and large sizes**.
   */
  icon?: IconName
  disabled?: boolean
  className?: string
  children: ReactNode
}

const Badge = ({
  variant = 'neutral',
  size = 'medium',
  prominence = 'default',
  icon,
  disabled = false,
  className,
  children,
}: BadgeProps) => {
  const theme = useTheme()

  /**
   * Badge should display an aria-label if the status is not neutral or primary
   */
  const ariaLabel = useMemo(
    () =>
      ['neutral', 'primary'].some(baseVariant => baseVariant === variant)
        ? undefined
        : variant,
    [variant],
  )

  const generatedStyles = useMemo(
    () => generateStyles({ prominence, theme }),
    [prominence, theme],
  )

  const sizeValue = SIZES[size]

  return (
    <StyledBox
      role="status"
      aria-label={ariaLabel}
      as="span"
      variant={disabled ? generatedStyles.disabled : generatedStyles[variant]}
      size={sizeValue}
      className={className}
    >
      {icon && sizeValue !== SIZES.small ? (
        <Icon name={icon} size={16} />
      ) : null}
      <StyledTypography
        fontSize={TEXT_SIZES[size]}
        fontWeight={500}
        color="inherit"
      >
        {children}
      </StyledTypography>
    </StyledBox>
  )
}

export default Badge
