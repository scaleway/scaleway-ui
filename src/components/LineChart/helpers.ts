import { Theme } from '@emotion/react'
import { DatumValue, Serie } from '@nivo/line'

const parse = (data?: DatumValue | null) => {
  if (typeof data === 'number') return data
  if (typeof data === 'string') return parseFloat(data)
  if (data instanceof Date) return data.getTime()

  return 0
}

export const getMaxChartValue = (preppedData?: Serie[]): number => {
  if (!preppedData?.length) return 0

  const maximum = Math.max(
    ...preppedData.map(data =>
      data.data.reduce(
        (max, { y }) => Math.max(parse(y), max),
        Number.NEGATIVE_INFINITY,
      ),
    ),
  )

  return Math.ceil(maximum + maximum * 0.1)
}

export const getMinChartValue = (preppedData?: Serie[]): number => {
  if (!preppedData?.length) return 0

  const minimum = Math.min(
    ...preppedData.map(data =>
      data.data.reduce(
        (min, { y }) => Math.min(parse(y), min),
        Number.POSITIVE_INFINITY,
      ),
    ),
  )

  return Math.floor(minimum - minimum * 0.1)
}

export const getMin = (values: number[] = []): number =>
  values.length ? Math.min(...values) : 0

export const getMax = (values: number[] = []): number =>
  values.length ? Math.max(...values) : 0

export const getAverage = (values: number[] = []): number =>
  values.length
    ? Math.round(
        (values.reduce((average, curr) => average + curr, 0) / values.length) *
          100,
      ) / 100
    : 0

export const getCurrent = (values: number[] = []): number =>
  values.length ? values[values.length - 1] : 0

export const getSelected = (
  label: string,
  index: number,
  selected: string[] = [],
): string[] => {
  const labelIndexed = label + index.toString()
  const found = selected.indexOf(labelIndexed)

  if (found > -1) {
    selected.splice(found, 1)
  } else {
    selected.push(labelIndexed)
  }

  return selected
}

export const legendColors = (theme: Theme): string[] =>
  [
    theme.colors.chartGreen,
    theme.colors.chartPurple,
    theme.colors.gray350,
  ] as string[]

export const getLegendColor = (index: number, theme: Theme): string => {
  const colors = legendColors(theme)

  return colors[index] || colors[colors.length - 1]
}
