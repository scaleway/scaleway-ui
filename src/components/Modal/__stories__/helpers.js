export const animations = [
  'fold',
  'zoom',
  'scaleUp',
  'scaleBack',
  'sketch',
  'slideTop',
  'slideBottom',
  'slideLeft',
  'slideRight',
]

export const widths = ['xxsmall', 'xsmall', 'small', 'medium', 'large']

export const placements = [
  'center',
  'top-left',
  'top',
  'top-right',
  'right',
  'bottom-right',
  'bottom',
  'bottom-left',
  'left',
]
