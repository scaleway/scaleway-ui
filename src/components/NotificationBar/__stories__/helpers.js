export default [
  'security',
  'info',
  'success',
  'warning',
  'warning-blue',
  'warning-orange',
  'alert-orange',
  'alert-red',
]
