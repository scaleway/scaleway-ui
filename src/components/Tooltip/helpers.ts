import { RefObject } from 'react'

export type TooltipPlacement = 'top' | 'right' | 'bottom' | 'left' | 'auto'
export const ARROW_WIDTH = 6 // in px
export const SPACE = 6 // in px
export const TOTAL_USED_SPACE = ARROW_WIDTH + SPACE // in px
export const DEFAULT_POSITIONS = {
  arrowLeft: 50,
  arrowTop: 99,
  arrowTransform: 'translate(-50%, -50)',
  left: 0,
  rotate: 135,
  tooltipInitialPosition: 'translate3d(0, 0, 0))',
  top: 0,
}

export const TOOLTIP_INITIAL_POSITION = {
  auto: `translate3d(0, ${TOTAL_USED_SPACE}px, 0)`,
  bottom: `translate3d(0, -${TOTAL_USED_SPACE}px, 0)`,
  left: `translate3d(${TOTAL_USED_SPACE}px, 0, 0)`,
  right: `translate3d(-${TOTAL_USED_SPACE}px, 0, 0)`,
  top: `translate3d(0, ${TOTAL_USED_SPACE}px, 0)`,
}

type ComputePlacementTypes = {
  childrenStructuredRef: DOMRect
  tooltipStructuredRef: DOMRect
}

/**
 * This function will find the best placement in a window for tooltip based on children position and tooltip size
 */
export const computePlacement = ({
  childrenStructuredRef,
  tooltipStructuredRef,
}: ComputePlacementTypes) => {
  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
  } = childrenStructuredRef

  const { width: tooltipWidth, height: tooltipHeight } = tooltipStructuredRef

  if (childrenTop - tooltipHeight - TOTAL_USED_SPACE < 0) {
    return 'bottom'
  }

  if (childrenLeft - tooltipWidth - TOTAL_USED_SPACE < 0) {
    return 'right'
  }

  if (childrenRight + tooltipWidth + TOTAL_USED_SPACE > window.innerWidth) {
    return 'left'
  }

  return 'top'
}

type ComputePositionsTypes = {
  placement: TooltipPlacement
  childrenRef: RefObject<HTMLDivElement>
  tooltipRef: RefObject<HTMLDivElement>
}

/**
 * This function will compute the positions of tooltip and arrow based on children position and tooltip size
 */
export const computePositions = ({
  placement,
  childrenRef,
  tooltipRef,
}: ComputePositionsTypes) => {
  const childrenStructuredRef = (
    childrenRef.current as HTMLDivElement
  ).getBoundingClientRect()
  const tooltipStructuredRef = (
    tooltipRef.current as HTMLDivElement
  ).getBoundingClientRect()

  const placementBasedOnWindowSize =
    placement === 'auto'
      ? computePlacement({
          childrenStructuredRef,
          tooltipStructuredRef,
        })
      : placement

  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
    width: childrenWidth,
    height: childrenHeight,
  } = childrenStructuredRef

  const { width: tooltipWidth, height: tooltipHeight } = tooltipStructuredRef

  switch (placementBasedOnWindowSize) {
    case 'bottom':
      return {
        arrowLeft: tooltipWidth / 2,
        arrowTop: -ARROW_WIDTH - 5,
        arrowTransform: '',
        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
        rotate: 180,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.bottom,
        top: childrenTop + childrenHeight + ARROW_WIDTH + SPACE,
      }
    case 'left':
      return {
        arrowLeft: tooltipWidth + ARROW_WIDTH + 5,
        arrowTop: tooltipHeight / 2,
        arrowTransform: 'translate(-50%, -50%)',
        left: childrenLeft - tooltipWidth - ARROW_WIDTH - SPACE * 2,
        rotate: -90,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.left,
        top: childrenTop - tooltipHeight / 2 + childrenHeight / 2,
      }
    case 'right':
      return {
        arrowLeft: -ARROW_WIDTH - 5,
        arrowTop: tooltipHeight / 2,
        arrowTransform: 'translate(50%, -50%)',
        left: childrenRight + ARROW_WIDTH + SPACE * 2,
        rotate: 90,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.right,
        top: childrenTop - tooltipHeight / 2 + childrenHeight / 2,
      }
    default: // top placement is default value
      return {
        arrowLeft: tooltipWidth / 2,
        arrowTop: tooltipHeight - 1,
        arrowTransform: '',
        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
        rotate: 0,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.top,
        top: childrenTop - tooltipHeight - ARROW_WIDTH - SPACE,
      }
  }
}
