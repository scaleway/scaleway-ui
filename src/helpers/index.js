export * from './clipboard'

export const parseIntOr = (str, fallback) => {
  const value = Number.parseInt(str, 10)
  return !Number.isNaN(value) ? value : fallback
}

export const onKeyOnlyNumbers = e => {
  // `e.which` and `e.keyCode` are deprecated (https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent)
  const keyCode = e.key.charCodeAt(0)
  if (keyCode < 48 || keyCode > 57) {
    e.preventDefault()
  }
}
