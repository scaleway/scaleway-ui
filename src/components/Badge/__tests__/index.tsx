import React from 'react'
import Badge, { badgeSizes, badgeVariants } from '..'
import { shouldMatchEmotionSnapshot } from '../../../helpers/jestHelpers'

describe('Badge', () => {
  test('renders correctly with default values', () =>
    shouldMatchEmotionSnapshot(<Badge>Sample badge</Badge>))

  badgeVariants.forEach(variant => {
    test(`renders correctly variant ${variant}`, () =>
      shouldMatchEmotionSnapshot(<Badge variant={variant}>Sample badge</Badge>))
  })

  badgeSizes.forEach(size => {
    test(`renders correctly size ${size}`, () =>
      shouldMatchEmotionSnapshot(<Badge size={size}>Sample badge</Badge>))
  })
})
