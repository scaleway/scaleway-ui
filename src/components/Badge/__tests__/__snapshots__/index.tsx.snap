// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly size medium 1`] = `
<DocumentFragment>
  .cache-rdc79q {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    class="egvl2gp0 cache-rdc79q"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly size rounded 1`] = `
<DocumentFragment>
  .cache-qt7dy3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  border-radius: 8px;
  font-size: 10px;
  height: 16px;
  padding: 2px 6px;
  text-transform: uppercase;
}

<span
    class="egvl2gp0 cache-qt7dy3"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly size small 1`] = `
<DocumentFragment>
  .cache-15uvhf0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  font-size: 12px;
  line-height: 24px;
  height: 24px;
}

<span
    class="egvl2gp0 cache-15uvhf0"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly size xsmall 1`] = `
<DocumentFragment>
  .cache-1jp69s7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  border-radius: 4px;
  font-size: 12px;
  line-height: 18px;
  height: 18px;
}

<span
    class="egvl2gp0 cache-1jp69s7"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly size xxsmall 1`] = `
<DocumentFragment>
  .cache-h73jke {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  border-radius: 4px;
  font-size: 10px;
  line-height: 16px;
  height: 16px;
  padding: 0 6px;
}

<span
    class="egvl2gp0 cache-h73jke"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant beta 1`] = `
<DocumentFragment>
  .cache-17vutqk {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ff8c69;
  color: #ffffff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="beta"
    class="egvl2gp0 cache-17vutqk"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant error 1`] = `
<DocumentFragment>
  .cache-16r5qw1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #dd3252;
  color: #ffffff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="error"
    class="egvl2gp0 cache-16r5qw1"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant info 1`] = `
<DocumentFragment>
  .cache-140mm8v {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #3f6ed8;
  color: #ffffff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="info"
    class="egvl2gp0 cache-140mm8v"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant light-beta 1`] = `
<DocumentFragment>
  .cache-dgtbtb {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #cc4e18;
  background-color: #ffefe6;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="beta"
    class="egvl2gp0 cache-dgtbtb"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant light-error 1`] = `
<DocumentFragment>
  .cache-lqz8xw {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #a6102d;
  background-color: #ffe1e7;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="error"
    class="egvl2gp0 cache-lqz8xw"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant light-info 1`] = `
<DocumentFragment>
  .cache-1qww9fs {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #1a4dbf;
  background-color: #e1ebff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="info"
    class="egvl2gp0 cache-1qww9fs"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant light-neutral 1`] = `
<DocumentFragment>
  .cache-dhdxq8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #4a4f62;
  background-color: #f6f6f8;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    class="egvl2gp0 cache-dhdxq8"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant light-primary 1`] = `
<DocumentFragment>
  .cache-b5hapk {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #390171;
  background-color: #eeeeff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    class="egvl2gp0 cache-b5hapk"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant light-success 1`] = `
<DocumentFragment>
  .cache-1dvyct7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #149a7b;
  background-color: #eafffb;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="success"
    class="egvl2gp0 cache-1dvyct7"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant neutral 1`] = `
<DocumentFragment>
  .cache-rdc79q {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    class="egvl2gp0 cache-rdc79q"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant primary 1`] = `
<DocumentFragment>
  .cache-y85sgz {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #4f0599;
  color: #ffffff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    class="egvl2gp0 cache-y85sgz"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant success 1`] = `
<DocumentFragment>
  .cache-13raz3x {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #22c8a2;
  color: #ffffff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="success"
    class="egvl2gp0 cache-13raz3x"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly variant warning 1`] = `
<DocumentFragment>
  .cache-16r5qw1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #dd3252;
  color: #ffffff;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    aria-label="warning"
    class="egvl2gp0 cache-16r5qw1"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;

exports[`Badge renders correctly with default values 1`] = `
<DocumentFragment>
  .cache-rdc79q {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 16px;
  padding: 0 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #151a2d;
  background-color: #d4dae7;
  font-size: 14px;
  line-height: 32px;
  height: 32px;
}

<span
    class="egvl2gp0 cache-rdc79q"
    role="status"
  >
    Sample badge
  </span>
</DocumentFragment>
`;
